### ISSUE TITLE
package adverbs are easy to lose if you use stubs

### ISSUE THREAD

**Issue by zoffixznet** [rakudo/rakudo#1769]
If you stub and apply the adverbs only on final definition, it's easy to end up without any of the values set:

```
05:10 | zostay | m: class F { ... }; class F:ver<1> { }; say F.^ver
-- | -- | --
05:10 | camelia | rakudo-moar 79ed89ba4: OUTPUT: «(Mu)␤»
05:10 | zostay | m: class F:ver<1> { ... }; class F { }; say F.^ver
05:10 | camelia | rakudo-moar 79ed89ba4: OUTPUT: «v1␤»
05:12 | zostay | apparently, it's important to make sure the first reference to a class as the adverbs set, even if it's a stub
```
https://irclog.perlgeek.de/perl6/2018-04-25#i_16088817

**Comment by lizmat** [comment]
Could this be related to the issue that “is repr” needs to be on the stub and cannot be set later?

**Comment by jnthn** [comment]
> Could this be related to the issue that “is repr” needs to be on the stub and cannot be set later?

No. That one is because we need to pick the REPR at the point we create a type object, and it's fixed from then on. And we need the type object to have a symbol to install.

By contrast, `ver` and `auth` are simply fields in the meta-object, and we can update them at any point. There's nothing guts-y there.
